op/model.py

```diff
@@ -72,6 +71,36 @@ def __init__(self, name, backend, cache):
 
         self.is_local = self.name == self._backend.local_app_name
 
+        self._status = None
+
+    @property
+    def status(self):
+        if not self.is_local:
+            return Unknown()
+
+        # TODO: do a leadership check (app status is not available to units for reading/writing).
+
+        if self._status:
+            return self._status
+
+        s = self._backend.status_get(True)
+        self._status = Status.from_string(s['status'], s['message'])
+
+        return self._status
+
+    @status.setter
+    def status(self, value):
+        if not isinstance(value, Status):
+            raise InvalidStatusError(f'Invalid value provided for unit {self} status: {value}')
+
+        if not self.is_local:
+            raise RuntimeError(f'Unable to get status for a non-local unit {self}')
```

Comment:
Message should say "set" instead of "get"

Reply:
Fixed in both places, thanks.